### DISCUSSION TITLE
Areas detect static bodies - per area3d switch

### DISCUSSION THREAD

**Discussion by smedelyan** [godot-jolt/godot-jolt#1032] [General]
Hi!
The following option is putting a significant overhead if turned on globally:
![image](https://github.com/user-attachments/assets/ba1485c2-ff71-4160-84f2-acf83101fddc)

Do you think it's possible to introduce this option on per-area basis? E.g. I still want this behavior ON for some areas (e.g. area-based projectiles), but for the rest of the world I'd better keep it OFF.

I understand that this is maybe completely depending on the Jolt itself, so I'm OK opening a discussion there instead

**Comment by mihe** [comment]
Hello!

> The following option is putting a significant overhead if turned on globally

Yes, this is mentioned in the project settings [documentation](https://github.com/godot-jolt/godot-jolt/blob/master/docs/settings.md#jolt-3d). The recommended mitigation strategy is to set up your collision layers/masks in such a way that most of your areas aren't able to detect overlaps with static bodies in the first place.

If you happen to have a very large `Area3D` (or many smaller instances of a certain `Area3D`) that's allowed to detect overlaps with complex collision shapes, then enabling this on a per-area basis won't do much good either way, unfortunately.

> I understand that this is maybe completely depending on the Jolt itself

While this is already toggleable on a per-area basis (or per-sensor rather) in Jolt itself, there is no way to expose such a property on `Area3D` from an extension at the moment. The only way to do so would be to introduce something like a `JoltArea3D` that inherits from `Area3D` and adds this new property, but since that would complicate moving back to Godot Physics (if there ever was a need) I chose not to go that route.

Note however that this extension has recently been merged into Godot itself as an engine module now (as mentioned in the [README](https://github.com/godot-jolt/godot-jolt?tab=readme-ov-file)) and the plan there is indeed for this setting to become a per-area property through the eventual support for physics engine modules/extensions to modify the properties of the default physics nodes, as mentioned under the "`Area3D` and static bodies" section of [the pull request](https://github.com/godotengine/godot/pull/99895). This will however not be backported to the extension, as this extension is now in maintenance mode, and will be deprecated/archived entirely at some later point.
